File: modules/@angular/platform-browser/src/web_workers/worker/worker_adapter.dart
library angular2.dom.webWorkerAdapter;

import 'abstract_html_adapter.dart';
import 'package:angular2/platform/common_dom.dart';

/**
 * This adapter is required to log error messages.
 *
 * Note: other methods all throw as the DOM is not accessible directly
 * in web worker context.
 */
class WebWorkerDomAdapter implements DomAdapter {
  static void makeCurrent() {
    setRootDomAdapter(new WebWorkerDomAdapter());
  }

  logError(error) {
    print('${error}');
  }

  log(error) {
    print('${error}');
  }

  logGroup(error) {
    print('${error}');
  }

  logGroupEnd() {}

  hasProperty(element, String name) {
    throw 'not implemented';
  }

  void setProperty(Element element, String name, Object value) =>
      throw 'not implemented';

  getProperty(Element element, String name) => throw 'not implemented';

  invoke(Element element, String methodName, List args) =>
      throw 'not implemented';

  get attrToPropMap => throw 'not implemented';

  set attrToPropMap(value) {
    throw 'readonly';
  }

  getGlobalEventTarget(String target) {
    throw 'not implemented';
  }

  getTitle() {
    throw 'not implemented';
  }

  setTitle(String newTitle) {
    throw 'not implemented';
  }

  String getEventKey(event) {
    throw 'not implemented';
  }

  void replaceChild(el, newNode, oldNode) {
    throw 'not implemented';
  }

  dynamic getBoundingClientRect(el) {
    throw 'not implemented';
  }

  Type getXHR() => throw 'not implemented';

  Element parse(String templateHtml) => throw 'not implemented';
  query(selector) {
    throw 'not implemented';
  }

  querySelector(el, String selector) {
    throw 'not implemented';
  }

  List querySelectorAll(el, String selector) {
    throw 'not implemented';
  }

  on(el, evt, listener) {
    throw 'not implemented';
  }

  Function onAndCancel(el, evt, listener) {
    throw 'not implemented';
  }

  dispatchEvent(el, evt) {
    throw 'not implemented';
  }

  createMouseEvent(eventType) {
    throw 'not implemented';
  }

  createEvent(eventType) {
    throw 'not implemented';
  }

  preventDefault(evt) {
    throw 'not implemented';
  }

  isPrevented(evt) {
    throw 'not implemented';
  }

  getInnerHTML(el) => throw 'not implemented';

  getOuterHTML(el) => throw 'not implemented';

  String nodeName(node) => throw 'not implemented';

  String nodeValue(node) => throw 'not implemented';

  String type(node) {
    throw 'not implemented';
  }

  content(node) => throw 'not implemented';

  firstChild(el) => throw 'not implemented';

  nextSibling(el) => throw 'not implemented';

  parentElement(el) => throw 'not implemented';

  List childNodes(el) => throw 'not implemented';
  List childNodesAsList(el) => throw 'not implemented';
  clearNodes(el) {
    throw 'not implemented';
  }

  appendChild(el, node) => throw 'not implemented';
  removeChild(el, node) {
    throw 'not implemented';
  }

  remove(el) => throw 'not implemented';
  insertBefore(el, node) {
    throw 'not implemented';
  }

  insertAllBefore(el, nodes) {
    throw 'not implemented';
  }

  insertAfter(el, node) {
    throw 'not implemented';
  }

  setInnerHTML(el, value) {
    throw 'not implemented';
  }

  getText(el) => throw 'not implemented';

  setText(el, String value) => throw 'not implemented';

  getValue(el) {
    throw 'not implemented';
  }

  setValue(el, String value) {
    throw 'not implemented';
  }

  getChecked(el) {
    throw 'not implemented';
  }

  setChecked(el, bool value) {
    throw 'not implemented';
  }

  createComment(String text) => throw 'not implemented';
  createTemplate(String html) => throw 'not implemented';
  createElement(tagName, [doc]) => throw 'not implemented';

  createElementNS(ns, tagName, [doc]) {
    throw 'not implemented';
  }

  createTextNode(String text, [doc]) => throw 'not implemented';

  createScriptTag(String attrName, String attrValue, [doc]) {
    throw 'not implemented';
  }

  createStyleElement(String css, [doc]) {
    throw 'not implemented';
  }

  createShadowRoot(el) {
    throw 'not implemented';
  }

  getShadowRoot(el) {
    throw 'not implemented';
  }

  getHost(el) {
    throw 'not implemented';
  }

  clone(node) => throw 'not implemented';
  getElementsByClassName(element, String name) {
    throw 'not implemented';
  }

  getElementsByTagName(element, String name) {
    throw 'not implemented';
  }

  List classList(element) => throw 'not implemented';

  addClass(element, String className) {
    throw 'not implemented';
  }

  removeClass(element, String className) {
    throw 'not implemented';
  }

  hasClass(element, String className) => throw 'not implemented';

  setStyle(element, String styleName, String styleValue) {
    throw 'not implemented';
  }

  bool hasStyle(Element element, String styleName, [String styleValue]) {
    throw 'not implemented';
  }

  removeStyle(element, String styleName) {
    throw 'not implemented';
  }

  getStyle(element, String styleName) {
    throw 'not implemented';
  }

  String tagName(element) => throw 'not implemented';

  attributeMap(element) => throw 'not implemented';

  hasAttribute(element, String attribute) => throw 'not implemented';

  hasAttributeNS(element, String ns, String attribute) {
    throw 'not implemented';
  }

  getAttribute(element, String attribute) => throw 'not implemented';

  getAttributeNS(element, String ns, String attribute) {
    throw 'not implemented';
  }

  setAttribute(element, String name, String value) {
    throw 'not implemented';
  }

  setAttributeNS(element, String ns, String name, String value) {
    throw 'not implemented';
  }

  removeAttribute(element, String attribute) {
    throw 'not implemented';
  }

  removeAttributeNS(element, String ns, String attribute) {
    throw 'not implemented';
  }

  templateAwareRoot(el) => throw 'not implemented';

  createHtmlDocument() {
    throw 'not implemented';
  }

  defaultDoc() {
    throw 'not implemented';
  }

  bool elementMatches(n, String selector) {
    throw 'not implemented';
  }

  bool isTemplateElement(Element el) => throw 'not implemented';

  bool isTextNode(node) => throw 'not implemented';
  bool isCommentNode(node) => throw 'not implemented';

  bool isElementNode(node) => throw 'not implemented';

  bool hasShadowRoot(node) {
    throw 'not implemented';
  }

  bool isShadowRoot(node) {
    throw 'not implemented';
  }

  importIntoDoc(node) {
    throw 'not implemented';
  }

  adoptNode(node) {
    throw 'not implemented';
  }

  String getHref(element) {
    throw 'not implemented';
  }

  void resolveAndSetHref(element, baseUrl, href) {
    throw 'not implemented';
  }

  List getDistributedNodes(Node) {
    throw 'not implemented';
  }

  bool supportsDOMEvents() => throw 'not implemented';

  bool supportsNativeShadowDOM() => throw 'not implemented';

  getHistory() => throw 'not implemented';

  getLocation() {
    throw 'not implemented';
  }

  getBaseHref() {
    throw 'not implemented';
  }

  resetBaseElement() {
    throw 'not implemented';
  }

  String getUserAgent() => throw 'not implemented';

  void setData(Element element, String name, String value) {
    throw 'not implemented';
  }

  getComputedStyle(element) {
    throw 'not implemented';
  }

  String getData(Element element, String name) => throw 'not implemented';

  // TODO(tbosch): move this into a separate environment class once we have it
  setGlobalVar(String name, value) {
    throw 'not implemented';
  }

  requestAnimationFrame(callback) {
    throw 'not implemented';
  }

  cancelAnimationFrame(id) {
    throw 'not implemented';
  }

  performanceNow() {
    throw 'not implemented';
  }

  getAnimationPrefix() {
    throw 'not implemented';
  }

  getTransitionEnd() {
    throw 'not implemented';
  }

  supportsAnimation() {
    throw 'not implemented';
  }
}

File: modules/@angular/platform-browser/src/web_workers/worker/worker_adapter.ts
import {Type} from '../../facade/lang';
import {DomAdapter, setRootDomAdapter} from '../../dom/dom_adapter';

/**
 * This adapter is required to log error messages.
 *
 * Note: other methods all throw as the DOM is not accessible directly in web worker context.
 */
export class WorkerDomAdapter extends DomAdapter {
  static makeCurrent() { setRootDomAdapter(new WorkerDomAdapter()); }

  logError(error) {
    if (console.error) {
      console.error(error);
    } else {
      console.log(error);
    }
  }

  log(error) { console.log(error); }

  logGroup(error) {
    if (console.group) {
      console.group(error);
      this.logError(error);
    } else {
      console.log(error);
    }
  }

  logGroupEnd() {
    if (console.groupEnd) {
      console.groupEnd();
    }
  }

  hasProperty(element, name: string): boolean { throw "not implemented"; }
  setProperty(el: Element, name: string, value: any) { throw "not implemented"; }
  getProperty(el: Element, name: string): any { throw "not implemented"; }
  invoke(el: Element, methodName: string, args: any[]): any { throw "not implemented"; }

  getXHR(): Type { throw "not implemented"; }

  get attrToPropMap(): {[key: string]: string} { throw "not implemented"; }
  set attrToPropMap(value: {[key: string]: string}) { throw "not implemented"; }

  parse(templateHtml: string) { throw "not implemented"; }
  query(selector: string): any { throw "not implemented"; }
  querySelector(el, selector: string): HTMLElement { throw "not implemented"; }
  querySelectorAll(el, selector: string): any[] { throw "not implemented"; }
  on(el, evt, listener) { throw "not implemented"; }
  onAndCancel(el, evt, listener): Function { throw "not implemented"; }
  dispatchEvent(el, evt) { throw "not implemented"; }
  createMouseEvent(eventType): any { throw "not implemented"; }
  createEvent(eventType: string): any { throw "not implemented"; }
  preventDefault(evt) { throw "not implemented"; }
  isPrevented(evt): boolean { throw "not implemented"; }
  getInnerHTML(el): string { throw "not implemented"; }
  getTemplateContent(el): any { throw "not implemented"; }
  getOuterHTML(el): string { throw "not implemented"; }
  nodeName(node): string { throw "not implemented"; }
  nodeValue(node): string { throw "not implemented"; }
  type(node): string { throw "not implemented"; }
  content(node): any { throw "not implemented"; }
  firstChild(el): Node { throw "not implemented"; }
  nextSibling(el): Node { throw "not implemented"; }
  parentElement(el): Node { throw "not implemented"; }
  childNodes(el): Node[] { throw "not implemented"; }
  childNodesAsList(el): Node[] { throw "not implemented"; }
  clearNodes(el) { throw "not implemented"; }
  appendChild(el, node) { throw "not implemented"; }
  removeChild(el, node) { throw "not implemented"; }
  replaceChild(el, newNode, oldNode) { throw "not implemented"; }
  remove(el): Node { throw "not implemented"; }
  insertBefore(el, node) { throw "not implemented"; }
  insertAllBefore(el, nodes) { throw "not implemented"; }
  insertAfter(el, node) { throw "not implemented"; }
  setInnerHTML(el, value) { throw "not implemented"; }
  getText(el): string { throw "not implemented"; }
  setText(el, value: string) { throw "not implemented"; }
  getValue(el): string { throw "not implemented"; }
  setValue(el, value: string) { throw "not implemented"; }
  getChecked(el): boolean { throw "not implemented"; }
  setChecked(el, value: boolean) { throw "not implemented"; }
  createComment(text: string): any { throw "not implemented"; }
  createTemplate(html): HTMLElement { throw "not implemented"; }
  createElement(tagName, doc?): HTMLElement { throw "not implemented"; }
  createElementNS(ns: string, tagName: string, doc?): Element { throw "not implemented"; }
  createTextNode(text: string, doc?): Text { throw "not implemented"; }
  createScriptTag(attrName: string, attrValue: string, doc?): HTMLElement {
    throw "not implemented";
  }
  createStyleElement(css: string, doc?): HTMLStyleElement { throw "not implemented"; }
  createShadowRoot(el): any { throw "not implemented"; }
  getShadowRoot(el): any { throw "not implemented"; }
  getHost(el): any { throw "not implemented"; }
  getDistributedNodes(el): Node[] { throw "not implemented"; }
  clone(node: Node): Node { throw "not implemented"; }
  getElementsByClassName(element, name: string): HTMLElement[] { throw "not implemented"; }
  getElementsByTagName(element, name: string): HTMLElement[] { throw "not implemented"; }
  classList(element): any[] { throw "not implemented"; }
  addClass(element, className: string) { throw "not implemented"; }
  removeClass(element, className: string) { throw "not implemented"; }
  hasClass(element, className: string): boolean { throw "not implemented"; }
  setStyle(element, styleName: string, styleValue: string) { throw "not implemented"; }
  removeStyle(element, styleName: string) { throw "not implemented"; }
  getStyle(element, styleName: string): string { throw "not implemented"; }
  hasStyle(element, styleName: string, styleValue?: string): boolean { throw "not implemented"; }
  tagName(element): string { throw "not implemented"; }
  attributeMap(element): Map<string, string> { throw "not implemented"; }
  hasAttribute(element, attribute: string): boolean { throw "not implemented"; }
  hasAttributeNS(element, ns: string, attribute: string): boolean { throw "not implemented"; }
  getAttribute(element, attribute: string): string { throw "not implemented"; }
  getAttributeNS(element, ns: string, attribute: string): string { throw "not implemented"; }
  setAttribute(element, name: string, value: string) { throw "not implemented"; }
  setAttributeNS(element, ns: string, name: string, value: string) { throw "not implemented"; }
  removeAttribute(element, attribute: string) { throw "not implemented"; }
  removeAttributeNS(element, ns: string, attribute: string) { throw "not implemented"; }
  templateAwareRoot(el) { throw "not implemented"; }
  createHtmlDocument(): HTMLDocument { throw "not implemented"; }
  defaultDoc(): HTMLDocument { throw "not implemented"; }
  getBoundingClientRect(el) { throw "not implemented"; }
  getTitle(): string { throw "not implemented"; }
  setTitle(newTitle: string) { throw "not implemented"; }
  elementMatches(n, selector: string): boolean { throw "not implemented"; }
  isTemplateElement(el: any): boolean { throw "not implemented"; }
  isTextNode(node): boolean { throw "not implemented"; }
  isCommentNode(node): boolean { throw "not implemented"; }
  isElementNode(node): boolean { throw "not implemented"; }
  hasShadowRoot(node): boolean { throw "not implemented"; }
  isShadowRoot(node): boolean { throw "not implemented"; }
  importIntoDoc(node: Node): Node { throw "not implemented"; }
  adoptNode(node: Node): Node { throw "not implemented"; }
  getHref(element): string { throw "not implemented"; }
  getEventKey(event): string { throw "not implemented"; }
  resolveAndSetHref(element, baseUrl: string, href: string) { throw "not implemented"; }
  supportsDOMEvents(): boolean { throw "not implemented"; }
  supportsNativeShadowDOM(): boolean { throw "not implemented"; }
  getGlobalEventTarget(target: string): any { throw "not implemented"; }
  getHistory(): History { throw "not implemented"; }
  getLocation(): Location { throw "not implemented"; }
  getBaseHref(): string { throw "not implemented"; }
  resetBaseElement(): void { throw "not implemented"; }
  getUserAgent(): string { throw "not implemented"; }
  setData(element, name: string, value: string) { throw "not implemented"; }
  getComputedStyle(element): any { throw "not implemented"; }
  getData(element, name: string): string { throw "not implemented"; }
  setGlobalVar(name: string, value: any) { throw "not implemented"; }
  requestAnimationFrame(callback): number { throw "not implemented"; }
  cancelAnimationFrame(id) { throw "not implemented"; }
  performanceNow(): number { throw "not implemented"; }
  getAnimationPrefix(): string { throw "not implemented"; }
  getTransitionEnd(): string { throw "not implemented"; }
  supportsAnimation(): boolean { throw "not implemented"; }
  supportsWebAnimation(): boolean { throw "not implemented"; }

  supportsCookies(): boolean { return false; }
  getCookie(name: string): string { throw "not implemented"; }
  setCookie(name: string, value: string) { throw "not implemented"; }
}
